### PR TITLE
Disable Create Project button if more than 6 projects

### PR DESCRIPTION
**Pull request by tylercloke**
### :nut_and_bolt: Description

Screen Shot 2019-05-01 at 4 12 10 PM

Screen Shot 2019-05-01 at 4 12 20 PM

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [ ] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/automate-ui/src/app/pages/project/list/project-list.component.html [hunk 1]
```diff
@@ -5,9 +5,12 @@
   <main>
     <chef-page-header>
       <chef-heading>Projects</chef-heading>
-      <chef-subheading>
+      <chef-subheading *ngIf="(sortedProjects$ | async)?.length < 6">
         Projects are named groups of resources used to define policies.
       </chef-subheading>
+      <chef-subheading *ngIf="(sortedProjects$ | async)?.length >= 6">
+        Projects group resources together. Max of 6 projects allowed.
```
Comment (msorens) [inline components/automate-ui/src/app/pages/project/list/project-list.component.html]: Please, please, please put that magic number used throughout into a named constant. Even on this line.

File: components/automate-ui/e2e/admin.e2e-spec.ts [hunk 2]
```diff
@@ -1533,6 +1533,43 @@ describe('Admin pages', () => {
         });
       });
     });
+
+    describe('when the max of 6 projects is reached', () => {
```
Comment (blakestier) [inline components/automate-ui/e2e/admin.e2e-spec.ts]: 6 custom projects, or 6 total projects? If it's custom, can we include that in the code wherever possible?

Comment (blakestier) [reply components/automate-ui/e2e/admin.e2e-spec.ts]: Or  . . . do we not display the chef-managed ones?

Comment (tylercloke) [reply components/automate-ui/e2e/admin.e2e-spec.ts]: Great question! Lemme look.

Comment (tylercloke) [reply components/automate-ui/e2e/admin.e2e-spec.ts]: Looks like they aren't included! Thanks for bringing that edge case up though :) 

https://github.com/chef/automate/blob/master/components/authz-service/server/v2/project.go#L202-L204

Comment (blakestier) [reply components/automate-ui/e2e/admin.e2e-spec.ts]: oh, right. The chef-managed one is just internal

File: components/automate-ui/src/app/pages/project/list/project-list.component.html [hunk 3]
```diff
@@ -17,7 +17,7 @@
       <ng-container *ngIf="(sortedProjects$ | async)?.length > 0">
         <chef-toolbar>
           <app-authorized [allOf]="['/iam/v2beta/projects', 'post']">
-            <chef-button id="create-button" primary (click)="openCreateModal()">Create Project</chef-button>
+            <chef-button [disabled]="(sortedProjects$ | async)?.length >= MAX_PROJECTS" id="create-button" primary (click)="openCreateModal()">Create Project</chef-button>
```
Comment (bcmdarroch) [inline components/automate-ui/src/app/pages/project/list/project-list.component.html]: @susanev this is a UX nitpick, but I think it's hard to immediately tell why the create button is disabled. I know it's in the sub-heading but my eye tends to skip over that. Some kind of "no more projects allowed" message to the right of the button would be helpful. (though I know there's supposed to be an `Update` Button there soon so I get it you don't want to mess with that design)

Comment (susanev) [reply components/automate-ui/src/app/pages/project/list/project-list.component.html]: I totally agree but unfortunately with ingest rules there are going to be other buttons to the right so we can't do that approach anymore. If we get user feedback on that from early customers we can find a better solution.

Comment (susanev) [reply components/automate-ui/src/app/pages/project/list/project-list.component.html]: Also, this limit should not be in place for more than a few months... hopefully.

**Pull request merged**